packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -483,30 +478,142 @@ export class PerformanceEvent {
     }
 }
 
+// Note - these Telemetry types should move to common-definitions package
+
+/**
+ * Broad classifications to be applied to individual properties as they're prepared to be logged to telemetry.
+ * Please do not modify existing entries for backwards compatibility.
+ */
+export enum TelemetryDataTag {
+    /** Non-sensitive data that can be logged plainly */
+    None,
+    /** Data containing terms from code packages that may have been dynamically loaded */
+    PackageData,
+    /** Personal data of a variety of classifications that pertains to the user */
+    UserData,
+}
+
+/**
+ * A property to be logged to telemetry containing both the value and the tag
+ */
+export interface ITaggedTelemetryPropertyType {
+    value: TelemetryEventPropertyType,
+    tag: TelemetryDataTag
+}
+
+/**
+ * Property bag containing a mix of value literals and wrapped values along with a tag
+ */
+export interface ITaggableTelemetryProperties {
+    [name: string]: TelemetryEventPropertyType | ITaggedTelemetryPropertyType;
+}
+
 /**
- * - Helper class for error tracking that can be used to log an error in telemetry.
- * - Care needs to be taken not to log PII information!
- * - This allows additional properties to be logged because object of this instance will record all of their properties
- *   when logged with a logger.
- * - Logger ignores all properties from any other error objects (not being instance of LoggingError), with exception of
- *   'message' & 'stack' properties if they exists on error object.
- * - In other words, logger logs only what it knows about and has good confidence it does not container PII information.
+ * Type guard to identify if a particular value appears to be a tagged telemetry property
  */
-export class LoggingError extends Error {
+export function IsTaggedTelemetryPropertyValue(val: any): val is ITaggedTelemetryPropertyType {
+    const valAsTagged: ITaggedTelemetryPropertyType = val;
```

Comment:
`Partial<ITaggedTelemetryPropertyType>` to enforce the `?.` below

Reply:
I'll check for truthiness first, I think that may resolve the issue better?  The `?.` not aligning with the type declaration definitely bugged me, thanks for calling it out.